src/services/tests/backend_url.rs

```diff
@@ -54,7 +54,7 @@ async fn core_uri_success_with_params() {
         exclude_spam
     );
 
-    assert_eq!(url.unwrap(), "http://mainnet-safe-transaction-web.safe.svc.cluster.local/api/v1/safes/0x1230B3d59858296A31053C1b8562Ecf89A2f888b/balances/usd/?trusted=false&exclude_spam=true".to_string());
+    assert_eq!(url.unwrap(), "https://safe-transaction.mainnet.gnosis.io/api/v1/safes/0x1230B3d59858296A31053C1b8562Ecf89A2f888b/balances/usd/?trusted=false&exclude_spam=true".to_string());
```

Comment:
This test is incorrect 🤔 we need to validate that we are using the vpc URL here, since this test asserts the URL formed to request against the core services. I am started to become more inclined towards having this as an env var. I will add another reason as a final comment on this PR :)

Reply:
I'm not sure if I understood why the test is incorrect. If it's being tested under a non optimised build then this should be the result. It might not provide the value that we want true but for that I'd look into adding a test for the optimised builds.

Reply:
Pushed a suggestion in 3c212e4

It works for both `cargo test` and `cargo test --release`

I added `--release` to the CI so that the release tests run.

Reply:
Coverage files are not generated this way. I will take a look ☝️

Reply:
Actually I prefer to not go the `cargo test --release` and revert the test changes. So it tests against the debug build. I'm wondering if I can test just these two tests with a specific profile (`release` profile) 🤔

Reply:
I will have a look at this maybe: https://github.com/actions-rs/grcov next week, in case something can be done regarding where the report is generated :)